src/redux/Cart/CartSlice.ts

```diff
@@ -0,0 +1,70 @@
+import { IcarData } from "./../../models/ICarData";
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+interface cartStatus {
+  items: {
+    item: IcarData;
+    quantity: number;
+  }[];
+  totalCount: number;
+}
+
+const initialState: cartStatus = {
+  items: [],
+  totalCount: 0,
+};
+
+export const CartSlice = createSlice({
+  name: "Cart",
+  initialState,
+  reducers: {
+    addToCart: (state, action: PayloadAction<IcarData>) => {
+      let itemExists = false;
+      if (state.items.length > 0) {
+        state.items.forEach((element) => {
+          if (element.item.id === action.payload.id) {
+            itemExists = true;
+            element.quantity += 1;
+          }
+        });
+      }
+      if (!itemExists) {
+        state.items.push({ item: action.payload, quantity: 1 });
+      }
+      state.totalCount = state.totalCount + 1;
+    },
+    increaseQuantity: (state, action: PayloadAction<IcarData>) => {
+      state.items.find(
+        (item) => item.item.id === action.payload.id
+      )!.quantity += 1;
```

Comment:
Honestly its a good compact way to write compact code, and works great for coding competitions as well 😉  and in this case its okay, but for production code its important to write clearely and define variables appropriately.
So I would like you to simplify the code and break it into proper variables and functions(if possible) such that the flow is apparent in one go